src/test/clojure/clara/test_rules.clj

```diff
@@ -3876,16 +3926,18 @@
               {:?temps [[] 20]} 1})
           (str "Two Cold facts with single firing for node type " node-type))
 
-      ;; FIXME: The correct assertion here is on equality to {:?temps [[] nil]}.
-      ;; Since (->Cold 10) is retracted we should use the initial value here.
-      ;; This is an existing defect that has been logged at https://github.com/rbrush/clara-rules/issues/188
       (is (= (-> empty-session
                  (insert (->Cold 10) (->Hot 100))
                  (fire-rules)
                  (retract (->Cold 10))
                  (fire-rules)
                  (query q1))
-             [{:?temps [[] 10]}])
+             (if (= "AccumulateWithJoinFilterNode" node-type)
+               [{:?temps [[] 10]}]
+               ;; FIXME: The correct assertion here is on equality to {:?temps [[] nil]}.
+               ;; Since (->Cold 10) is retracted we should use the initial value here.
+               ;; This is an existing defect that has been logged at https://github.com/rbrush/clara-rules/issues/188
```

Comment:
I should have mentioned this in my synopsis of the changes.  
Where I wrote
> Addresses the issues 188, 189, and 200 for AccumulateNode

I specifically meant those issues were addressed for AccumulateNode and not AccumulateWithJoinFilterNode.  I should have made that more obvious though.  The issues #188 and #189 should remain open until we fix that one as well.

I left AccumulateWithJoinFilterNode out of this review since there were already significant changes and I figured it would be more of a cohesive set of changes if I kept it contained to the AccumulateNode for now.  

I think much of these same patterns can be used by the AccumulateWithJoinFilterNode to accomplish the same goals.  I'd hope and like to figure out a little bit more of a pattern of reuse of this logic when doing that, since the logic has became more complicated over time for both nodes.  There are some significant differences between the two nodes still, but I think we can get some reuse around the handling of the `:initial-value`.  This is why I considered that to be too much more work to add to the scope of these changes.